app/src/main/java/com/github/sdp/tarjetakuna/model/User.kt

```diff
@@ -1,25 +1,52 @@
 package com.github.sdp.tarjetakuna.model
 
+import com.github.sdp.tarjetakuna.database.CardPossession
 import com.github.sdp.tarjetakuna.database.DBMagicCard
+import com.github.sdp.tarjetakuna.database.UserRTDB
+import com.google.firebase.database.DataSnapshot
+import java.util.concurrent.CompletableFuture
 
 /**
  * Represents a user.
  */
 data class User(
-    val email: String,
     val username: String,
     val cards: List<DBMagicCard>,
     val location: Coordinates
+    //val chats: List<Chat>
+
 ) {
     init {
-        require(email.isNotBlank()) { "Email cannot be blank" }
         require(
-            email.matches(
+            username.matches(
                 Regex(
                     "^[a-zA-Z0-9]+(?:\\.[a-zA-Z0-9]+)*@[a-zA-Z0-9]+(?:\\.[a-zA-Z0-9]+)*\$"
                 )
             )
         ) { "Email is not valid" }
-        require(username.isNotBlank()) { "Username cannot be blank" }
     }
+
+    private val userRTDB = UserRTDB()
+    fun getAllCards(username: String = this.username): CompletableFuture<DataSnapshot> {
+        //todo
+        return userRTDB.getAllCardsFromUserCollection()
```

Comment:
wrap around database or firebase interface, and those will hold the database reference